app/actions/remote/groups.ts

```diff
@@ -1,104 +1,104 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
+import {storeGroups} from '@actions/local/group';
+import {prepareGroups} from '@app/queries/servers/group';
 import {Client} from '@client/rest';
+import DatabaseManager from '@database/manager';
 import NetworkManager from '@managers/network_manager';
 
 import {forceLogoutIfNecessary} from './session';
 
```

Comment:
Just notice that all the network clients in this file are not bounded by a try-catch.  It is not guaranteed that you get one each time.

```
  let client: Client;
    try {
        client = NetworkManager.getClient(serverUrl);
    } catch (error) {
        return {error};
    }
```

Reply:
I'm not sure I follow; how is that different to

```
try {
    const client: Client = NetworkManager.getClient(serverUrl);
    return client.doSomething();
catch (error) {
    return {error};
}
```

I don't use the `client` beyond the the initial `try`, so I don't quite see why I should have it as `let client: Client;` outside the try catch.

If I want the `client` bound in its own try / catch; the code is a bit repetitive no?

```
let client: Client;
try {
    client = NetworkManager.getClient(serverUrl);
catch (error) {
    return {error};
}
try {
    const response = client.doSomething();
catch (error) {
    return {error};
}

return response;

```

Reply:
My bad...there is some try-catch in this file... you're good to go :)